tests/src/PatternAssertions/HighlightedListAssert.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\PatternAssertions;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Assertions for the Highlighted list pattern.
+ *
+ * @see ./templates/patterns/highlighted_list/highlighted_list.ui_patterns.yml
+ */
+class HighlightedListAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions($variant): array {
+    return [
+      'title' => [
+        [$this, 'assertElementText'],
+        'div#highlighted-news-block h2.ecl-u-type-heading-2',
+      ],
+      'highlighted_item' => [
+        [$this, 'assertHighlightedItem'],
+        'div#highlighted-news-block div.ecl-row div.ecl-col-l-8.ecl-u-d-flex.ecl-u-flex-column',
+      ],
+      'items' => [
+        [$this, 'assertItems'],
+        'div.highlighted-news-divider div.ecl-row div.ecl-col-l-4.ecl-u-d-flex.ecl-u-flex-column',
+      ],
+      'see_more_label' => [
+        [$this, 'assertElementText'],
+        'div#highlighted-news-block > div.ecl-u-mt-s a.ecl-link.ecl-link--standalone .ecl-link__label',
+      ],
+      'see_more_url' => [
+        [$this, 'assertElementAttribute'],
+        'div#highlighted-news-block > div.ecl-u-mt-s a.ecl-link.ecl-link--standalone',
+        'href',
+      ],
+      'detail' => [
+        [$this, 'assertDetail'],
+        'div#highlighted-news-block div.ecl.ecl-u-mt-s',
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertBaseElements(string $html, string $variant): void {
+    $crawler = new Crawler($html);
+    $highlighted_list = $crawler->filter('div#highlighted-news-block');
+    self::assertCount(1, $highlighted_list);
+  }
+
+  /**
+   * Asserts the Highlighted item of the list.
+   *
+   * @param array|null $expected_highlighted_item
+   *   The expected item array (image array, title, url, primary meta array
+   *   and secondary meta).
+   * @param string $selector
+   *   The CSS selector to find the item.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertHighlightedItem(?array $expected_highlighted_item, string $selector, Crawler $crawler) {
+    if (is_null($expected_highlighted_item)) {
+      $this->assertElementNotExists($selector, $crawler);
+      return;
+    }
+    $this->assertElementExists($selector, $crawler);
+    $highlighted_wrapper = $crawler->filter($selector);
+    $list_item_assert = new ListItemAssert();
+    $html = $highlighted_wrapper->html();
+    $list_item_assert->assertPattern($expected_highlighted_item, $html);
```

Comment:
Asserting the ListItemAssert doesn;t work out of the box. I think we have a problem in ` ListItemAssert::assertThumbnailImage()`:

IMO, this:

```
 $variant_class = $variant === 'thumbnail_primary' ? 'picture.ecl-content-item__picture--large.ecl-content-item__picture--left' : 'picture.ecl-content-item__picture--large.ecl-content-item__picture--right';
```

should become this:

```
    $variant_class = 'picture.ecl-content-item__picture--large.ecl-content-item__picture--top';
    switch ($variant) {
      case 'thumbnail_primary':
        $variant_class = 'picture.ecl-content-item__picture--large.ecl-content-item__picture--left';
        break;
      case 'thumbnail_secondary':
        $variant_class = 'picture.ecl-content-item__picture--large.ecl-content-item__picture--right';
        break;
    }
```

However, I noticed a problem. For the second item (the top one in the right column), you process it to show the image as `medium` so the class changes to `ecl-content-item__picture--large` meaning the above will fail. Locally I removed that class from the assertion and now it works. Do you have a better way? I guess we cannot pass dynamically..

Reply:
We should be passing it dynamically.